PixiEditor/Views/UserControls/Zoombox.xaml.cs

```diff
@@ -0,0 +1,274 @@
+﻿using System;
+using System.ComponentModel;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using System.Windows.Markup;
+
+namespace PixiEditor.Views.UserControls
+{
+    [ContentProperty(nameof(AdditionalContent))]
+    public partial class Zoombox : ContentControl, INotifyPropertyChanged
+    {
+        public enum Mode
+        {
+            Normal, MoveTool, ZoomTool
+        }
+
+        private interface IDragOperation
+        {
+            void Start(MouseButtonEventArgs e);
+
+            void Update(MouseEventArgs e);
+
+            void Terminate();
+        }
+
+        private class MoveDragOperation : IDragOperation
+        {
+            private Zoombox parent;
+            private Point prevMousePos;
+
+            public MoveDragOperation(Zoombox zoomBox)
+            {
+                parent = zoomBox;
+            }
+            public void Start(MouseButtonEventArgs e)
+            {
+                prevMousePos = e.GetPosition(parent.mainCanvas);
+                parent.mainCanvas.CaptureMouse();
+            }
+
+            public void Update(MouseEventArgs e)
+            {
+                var curMousePos = e.GetPosition(parent.mainCanvas);
+                parent.SpaceOriginPos += curMousePos - prevMousePos;
+                prevMousePos = e.GetPosition(parent.mainCanvas);
```

Comment:
Wouldn't `curMousePos` already be assigned to the correct point

Reply:
I guess